Betriebsstundenzaehler/module.php

```diff
@@ -44,26 +47,19 @@ public function ApplyChanges()
     {
         //Never delete this line!
         parent::ApplyChanges();
-        $newStatus = 102;
 
-        if (!$this->ReadPropertyBoolean('Active')) {
-            $newStatus = 104;
-        } else {
-            $newStatus = $this->getErrorState();
-        }
-        $this->SetStatus($newStatus);
-        if ($newStatus != 102) {
-            $this->SetTimerInterval('UpdateCalculationTimer', 0);
-            $this->SetValue('OperatingHours', 0);
-            return;
+        //Only call this in READY state. On startup the ArchiveControl instance might not be available yet
+        if (IPS_GetKernelRunlevel() == KR_READY) {
+            $this->setupInstance();
         }
+    }
 
-        if ($this->ReadPropertyBoolean('Active') && $this->GetTimerInterval('UpdateCalculationTimer') < ($this->ReadPropertyInteger('Interval') * 1000 * 60)) {
-            $this->SetTimerInterval('UpdateCalculationTimer', $this->ReadPropertyInteger('Interval') * 1000 * 60);
-        } elseif (!$this->ReadPropertyBoolean('Active')) {
-            $this->SetTimerInterval('UpdateCalculationTimer', 0);
+    public function MessageSink($TimeStamp, $SenderID, $Message, $Data)
+    {
+        //Calculate when the archive module is loaded
+        if ($Message == IPS_KERNELMESSAGE && $Data[0] == KR_READY) {
+            $this->setupInstance();
         }
-        $this->Calculate();
```

Comment:
Ist das $this->Calculate() mittlerweile verschwunden? Denn das steckt scheinbar nicht in setupInstance()